sdk/storage/azure-storage-blob/azure/storage/blob/models.py

```diff
@@ -768,7 +768,7 @@ class BlobBlock(DictMixin):
         Block size in bytes.
     """
 
-    def __init__(self, block_id=None, state=BlockState.Latest):
+    def __init__(self, block_id, state=BlockState.Latest):
```

Comment:
If you change this you have to update the `from_generated` classmethod on line 777 below to pass in the required param